packages/blade/src/components/Carousel/Carousel.web.tsx

```diff
@@ -454,21 +446,33 @@ const Carousel = ({
     },
   );
 
+  // set initial active slide on mount
+  useIsomorphicLayoutEffect(() => {
+    if (!id) return;
+    goToSlideIndex(activeSlide);
+    scrollToSlide(activeSlide, false);
+  }, [id]);
```

Comment:
when does the id change? wouldn't it always stay same 🤔

Reply:
useId hook returns null on first render, thus we only want to run it once id is actually present.